### ISSUE TITLE
Display video/bookshelf original URL (or canonical URL if possible!) to all users [ASIDE: upload of offline video works, despite red error "Failed to Move Cover File" in web UI]

### ISSUE THREAD

**Issue by avni** [iiab/calibre-web#107]
**Is your feature request related to a problem? Please describe.**
After I downloaded a video, I could not easily find the source URL from the Books or Edit Metadata page.

**Describe the solution you'd like**
When viewing or editing a downloaded media, in addition to publisher, it would be helpful to see the source URL.  This can be for testing purposes (my use case), but I can also see educators wanting to view source URL to verify veracity or accuracy of content. 
e.g., https://www.youtube.com/watch?v=OnTYLyNUPMc

**Describe alternatives you've considered**
I can see the source URL on the Tasks page, but this requires me inferring which video URL refers to which downloaded video available on the Books page. Also I expect that some educators using an IIAB install may not be Admins and hence not able to access the Tasks page.

**Additional context**
These are the 2 pages where seeing the source URL will be helpful. You could make it so that the field is not editable on the edit page if you want to restrict changing the URL after downlaoding. 

<img width="903" alt="Screenshot 2024-01-20 at 6 24 18 PM" src="https://github.com/iiab/calibre-web/assets/47825/e7bab559-8999-422b-abd2-e482995c9837">
<img width="1098" alt="Screenshot 2024-01-20 at 6 24 27 PM" src="https://github.com/iiab/calibre-web/assets/47825/56e69f51-7aa0-4476-bec6-020eb592b22a">

**Comment by holta** [comment]
Definitely!

Other metadata is also urgently needed by educators & students too.  Possibly including things like:

- HH:MM:SS duration of video
- Date of publication
- Popularity
  - e.g. average YouTube Views-per-Day
  - e.g. offline community Total Views

**Comment by holta** [comment]
@deldesir can we make this happen in June?

(Or if not yet, what higher-priority prep work needs to happen first?)

**Comment by deldesir** [comment]
No problem. Let's plan this for next week.

**Comment by holta** [comment]
> To Do:
>
>1) Many curriculum devs (parents, tutors, teachers, etc) will strongly prefer `Source: URL` on the very bottom of the description.
>
>   To reduce distraction.
>
>   “Offline First” is not just a slogan: this needs to be tried (soon).
>
> 2) Other curatorial metadata like HH:MM:SS, e.g. video duration, will need to be surfaced far more prominently (for educators, kids, etc) in coming months — as summarized here:
>
>    https://github.com/iiab/calibre-web/issues/107#issuecomment-1902439943

(Pasted in from...)
- https://github.com/iiab/calibre-web/pull/183#issuecomment-2166928357

**Comment by holta** [comment]
Regarding this PR just merged...

- PR #185

...a question for @deldesir:

> 1) I inserted the word "Internet" within "Original Internet URL" to help educators/kids understand that the original video came from the **ONline** Internet...
>
> 2) But what happens with this code when the original video comes from an **OFFline** (local, grassroots community) video — _do you know?_

**Comment by holta** [comment]
> > 2. But what happens with this code when the original [was] an **OFFline** (local, grassroots community) video — _do you know?_
>
> An offline resource shared would be better identifiable by the the publisher name IMO. Could be an individual, an institution...

Here's the technical answer to my question above — this is exactly what happened when I tried to upload 2.8MB "sample-5s - Unknown.mp4" to LRN2:

> Failed to Move Cover File /library/calibre-web/Unknown/sample-5s - Unknown (1000): [Errno 2] No such file or directory: '/tmp/calibre_web/97da78f4fcd9095d11967496de38b04e.cover.jpg'

![image](https://github.com/iiab/calibre-web/assets/2458907/329a2ada-2f31-4172-a493-276f50efa5c5)

![image](https://github.com/iiab/calibre-web/assets/2458907/9b49ff33-7abb-4b53-a67b-13ff8af3f986)

![image](https://github.com/iiab/calibre-web/assets/2458907/8943c2a4-0480-4c74-ba5e-975b2bc398e9)

In short, the process is rather ugly: (but at least it does work in the end, and "Original Internet URL:" field does not appear, which is the correct behavior!)

https://lrn2.org/live/cover/1000/sm?c=1718393846
https://lrn2.org/live/download/1000/mp4/1000.mp4
https://lrn2.org/live/read/1000/mp4

**Comment by holta** [comment]
> [*] Or possibly even original OFFline photo??

Upload of an OFFline photo works!  Example of an 86.7 KB .WEBP photo upload:

(screenshot of https://lrn2.org/live/ below)
![image](https://github.com/iiab/calibre-web/assets/2458907/2213ac26-4d01-4b10-8e0d-24fcc234fcaa)

(screenshot of https://lrn2.org/live/book/1001 below)
![image](https://github.com/iiab/calibre-web/assets/2458907/4f5b62ce-f134-4171-b691-76cbfa4ab826)

(screenshot of https://lrn2.org/live/admin/book/1001 below)
![image](https://github.com/iiab/calibre-web/assets/2458907/a9216390-e5cf-4c08-97f0-3a9af1fee19d)

**Comment by deldesir** [comment]
I will work on this. I'll use ffmpeg directly since our tests with https://github.com/smartcrop/smartcrop.py were not conclusive.

**Comment by holta** [comment]
> what happened when I tried to upload 2.8MB OFFline video "sample-5s - Unknown.mp4" to LRN2:
>
> > Failed to Move Cover File /library/calibre-web/Unknown/sample-5s - Unknown (1000): [Errno 2] No such file or directory: '/tmp/calibre_web/97da78f4fcd9095d11967496de38b04e.cover.jpg'

Progress — I'm hoping this PR can be merged today — if it's confirmed to be solid:

- PR #196

**Comment by holta** [comment]
A bookshelf's original URL and key context (description etc!?) should also definitely be displayed for everyone (eventually!)

Right now bookshelves unfortunately don't provide any "lineage" details or "structural" details[*] except for the bookshelf title — as can be seen here:

![image](https://github.com/iiab/calibre-web/assets/2458907/da6d2b4a-f370-4a84-a0c5-4941a4ec8bc8)
Screenshot from: https://lrn2.org/live/shelf/44

[*] e.g. what is the average HH:MM:SS of every video on the shelf — to help all teachers/parents very quickly organize their lesson plan?

**Comment by holta** [comment]
Just FYI ugly HTML `<br><br>` tags are displayed, concatenated on the bottom of video description in many-if-not-most cases:

```
<br><br>Original Internet URL: ...
```

Not an emergency and not the end of the world.  But this should really be fixed.  (Users should see "Original Internet URL: ..." on a completely separate line.)

Related:

- PR #183
- PR #185

**Comment by deldesir** [comment]
The regression is fixed by upstream commit https://github.com/janeczku/calibre-web/commit/551828f8cf8fb9f052d90456a3eb7ebaa8529cc6
